src/qpx_mmc_css.cpp

```diff
@@ -0,0 +1,1909 @@
+
+  /* Now the actual blocks doing the encryption.  Each
+   * of these works on 40 bits at a time and are quite
+   * similar.
+   */
+  i_index = 0;
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_scratch[i]) {
+    i_index = p_bits[25 + i] ^ p_scratch[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+
+    p_tmp1[i] = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+  }
+  p_tmp1[4] ^= p_tmp1[0];
+
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_tmp1[i]) {
+    i_index = p_bits[20 + i] ^ p_tmp1[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+
+    p_tmp2[i] = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+  }
+  p_tmp2[4] ^= p_tmp2[0];
+
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_tmp2[i]) {
+    i_index = p_bits[15 + i] ^ p_tmp2[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+    i_index = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+
+    p_tmp1[i] = p_crypt_tab0[i_index] ^ p_crypt_tab2[i_index];
+  }
+  p_tmp1[4] ^= p_tmp1[0];
+
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_tmp1[i]) {
+    i_index = p_bits[10 + i] ^ p_tmp1[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+
+    i_index = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+
+    p_tmp2[i] = p_crypt_tab0[i_index] ^ p_crypt_tab2[i_index];
+  }
+  p_tmp2[4] ^= p_tmp2[0];
+
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_tmp2[i]) {
+    i_index = p_bits[5 + i] ^ p_tmp2[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+
+    p_tmp1[i] = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+  }
+  p_tmp1[4] ^= p_tmp1[0];
+
+  for (i = 5, i_term = 0; --i >= 0; i_term = p_tmp1[i]) {
+    i_index = p_bits[i] ^ p_tmp1[i];
+    i_index = p_crypt_tab1[i_index] ^ ~p_crypt_tab2[i_index] ^ i_cse;
+
+    p_key[i] = p_crypt_tab2[i_index] ^ p_crypt_tab3[i_index] ^ i_term;
+  }
+
+  return;
+}
+
+/*****************************************************************************
+ * DecryptKey: decrypt p_crypted with p_key.
+ *****************************************************************************
+ * Used to decrypt the disc key, with a player key, after requesting it
+ * in _dvdcss_disckey and to decrypt title keys, with a disc key, requested
+ * in _dvdcss_titlekey.
+ * The player keys and the resulting disc key are only used as KEKs
+ * (key encryption keys).
+ * Decryption is slightly dependant on the type of key:
+ *  -for disc key, invert is 0x00,
+ *  -for title key, invert if 0xff.
+ *****************************************************************************/
+static void css_DecryptKey(unsigned char invert, unsigned char const *p_key,
+                           unsigned char const *p_crypted,
+                           unsigned char *p_result) {
+  unsigned int i_lfsr1_lo;
+  unsigned int i_lfsr1_hi;
+  unsigned int i_lfsr0;
+  unsigned int i_combined;
+  unsigned char o_lfsr0;
+  unsigned char o_lfsr1;
```

Comment:
Codacy found an issue: [The scope of the variable 'o_lfsr1' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)